exercises/tree-building/examples/success-standard/src/TreeBuilding.hs

```diff
@@ -0,0 +1,77 @@
+module TreeBuilding (newTree, Record(..), Tree(..)) where
+
+import Data.List
+import Data.Maybe (isNothing)
+
+type Id = Int
+type Children = [Tree]
+type ParentGrouping = (Maybe Id, [Id])
+
+data Record = Record Id (Maybe Id) deriving (Eq, Show)
```

Comment:
Since you created these two as helper functions, you might as well create them as record fields instead.

```suggestion
data Record = Record
  { recordId     :: Id
  , recordParent :: Maybe Id
  } deriving (Eq, Show)
```